pkg/brokers/rabbitmq/broker.go

```diff
@@ -164,12 +164,14 @@ func (r *RabbitMQBroker) Dequeue(ctx context.Context, queue string) (interfaces.
 	}
 
 	// Store delivery tag for ACK/NACK
-	if rmqJob, ok := job.(*RMQJob); ok {
```

Comment:
Learning you should take from this is in Go you should be mindful of error handling. Generally we always do:
```go
var rmqJob *RMQJob
if rmqJob, ok := job.(*RMQJob); !ok {
  // Log the error
  // Maybe exit with:
  // return nil, errors.New('')
end

rmqJob.deliveryTag = delivery.DeliveryTag
rmqJob.channel = r.channel

return rmqJob, nil
```

But I see that here the error was something else.

Reply:
Use guard clauses in go and you will never miss an error :)